The scheduler unit test "only one workload can borrow one resources from the same flavor..." flakes

/kind bug
/kind flake

**What happened**:

Failed unit test: `sigs.k8s.io/kueue/pkg/scheduler: TestSchedule/only_one_workload_can_borrow_one_resources_from_the_same_flavor_in_the_same_cycle_if_cohort_quota_cannot_fit ` with output:

```
  scheduler_test.go:1120: Unexpected scheduled workloads (-want,+got):
          map[string]v1beta1.Admission{
        - 	"sales/wl1": {
        - 		ClusterQueue: "cq1",
        - 		PodSetAssignments: []v1beta1.PodSetAssignment{
        - 			{
        - 				Name:          "main",
        - 				Flavors:       map[v1.ResourceName]v1beta1.ResourceFlavorReference{...},
        - 				ResourceUsage: v1.ResourceList{...},
        - 				Count:         &1,
        - 			},
        - 		},
        - 	},
        + 	"sales/wl2": {
        + 		ClusterQueue: "cq2",
        + 		PodSetAssignments: []v1beta1.PodSetAssignment{
        + 			{
        + 				Name:          "main",
        + 				Flavors:       map[v1.ResourceName]v1beta1.ResourceFlavorReference{...},
        + 				ResourceUsage: v1.ResourceList{...},
        + 				Count:         &1,
        + 			},
        + 		},
        + 	},
          }
    scheduler_test.go:1145: Unexpected assigned clusterQueues in cache (-want,+got):
          map[string]v1beta1.Admission{
        - 	"sales/wl1": {
        - 		ClusterQueue: "cq1",
        - 		PodSetAssignments: []v1beta1.PodSetAssignment{
        - 			{
        - 				Name:          "main",
        - 				Flavors:       map[v1.ResourceName]v1beta1.ResourceFlavorReference{...},
        - 				ResourceUsage: v1.ResourceList{...},
        - 				Count:         &1,
        - 			},
        - 		},
        - 	},
        + 	"sales/wl2": {
        + 		ClusterQueue: "cq2",
        + 		PodSetAssignments: []v1beta1.PodSetAssignment{
        + 			{
        + 				Name:          "main",
        + 				Flavors:       map[v1.ResourceName]v1beta1.ResourceFlavorReference{...},
        + 				ResourceUsage: v1.ResourceList{...},
        + 				Count:         &1,
        + 			},
        + 		},
        + 	},
          }
    scheduler_test.go:1150: Unexpected elements left in the queue (-want,+got):
          map[string]sets.Set[string]{
        + 	"cq1": {"sales/wl1": {}},
        - 	"cq2": {"sales/wl2": {}},
          }
--- FAIL: TestSchedule/only_one_workload_can_borrow_one_resources_from_the_same_flavor_in_the_same_cycle_if_cohort_quota_cannot_fit (0.07s)
```

**What you expected to happen**:

No failure

**How to reproduce it (as minimally and precisely as possible)**:

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/1405/pull-kueue-test-unit-main/1735227794121560064

**Anything else we need to know?**:

I think this is the second time I see it

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

--- a/pkg/scheduler/scheduler_test.go
+++ b/pkg/scheduler/scheduler_test.go
@@ -1289,15 +1289,17 @@ func TestEntryOrdering(t *testing.T) {
 			wantOrder:       []string{"old", "recently_evicted", "new", "new_high_pri", "old_borrowing", "evicted_borrowing", "high_pri_borrowing", "new_borrowing"},
 		},
 	} {
-		features.SetFeatureGateDuringTest(t, features.PrioritySortingWithinCohort, tc.prioritySorting)
-		sort.Sort(entryOrdering(input))
-		order := make([]string, len(input))
-		for i, e := range input {
-			order[i] = e.Obj.Name
-		}
-		if diff := cmp.Diff(tc.wantOrder, order); diff != "" {
-			t.Errorf("%s: Unexpected order (-want,+got):\n%s", tc.name, diff)
-		}
+		t.Run(tc.name, func(t *testing.T) {
+			t.Cleanup(features.SetFeatureGateDuringTest(t, features.PrioritySortingWithinCohort, tc.prioritySorting))
+			sort.Sort(entryOrdering(input))
+			order := make([]string, len(input))
+			for i, e := range input {
+				order[i] = e.Obj.Name
+			}
+			if diff := cmp.Diff(tc.wantOrder, order); diff != "" {
+				t.Errorf("%s: Unexpected order (-want,+got):\n%s", tc.name, diff)
+			}
+		})
 	}
 }
 

